Week3/homework/README.md

```diff
@@ -0,0 +1,47 @@
+## **TODO API**
+
+> < table > => users | todos | todo_lists | categories
+
+#### **Basic Crud Operations**
+
+| Request      | Endpoints          | Description        |
+| ------------ | ------------------ | :----------------- |
+| **`GET`**    | `/api/<table>`     | Get all items      |
+| **`GET`**    | `/api/<table>/:id` | Get item by id     |
+| **`POST`**   | `/api/<table>`     | Create single item |
+| **`PUT`**    | `/api/<table>/:id` | Update item        |
+| **`DELETE`** | `/api/<table>/:id` | Delete item by id  |
+
+#### **Multiple Crud Operations**
+
+| Request      | Endpoints                   | Description                                                                                                                     |
+| ------------ | --------------------------- | :------------------------------------------------------------------------------------------------------------------------------ |
+| **`POST`**   | `/api/<table>/multiple`     | Create multiple users array<br>You must provide array with objects which contain custom properties                              |
+| **`DELETE`** | `/api/<table>/multiple/:id` | Delete users by multiple ids<br>Id numbers must be seperated by `|`<br>For example: `DELETE` => `/api/todos/multiple/1 | 2 | 4` |
+
+
+#### **Custom Operations**
+
+| Request   | Endpoints                  | Description                       |
+| --------- | -------------------------- | :-------------------------------- |
+| **`GET`** | `/api/user/:id/todos`      | Get all todos of a user           |
```

Comment:
this is great. It is very common to see such patterns in rest where you can get a list of things for another object by following a route. It feels very natural 👍